### ISSUE TITLE
Node controller is not marking nodes as NotReady and not removing them

### ISSUE THREAD

**Issue by jszczepkowski** [kubernetes/kubernetes#39723]
**Kubernetes version** (use `kubectl version`):
v1.6.0-alpha.0.2942+e827393fa2335c-dirty

**Environment**:
GCE, gci

**What happened**:
Node controller is not marking nodes as NotReady and not removing them. After a worker node or a master replica is removed, it is still marked as a ready node.

```
$ gcloud compute instances list
NAME                            ZONE            MACHINE_TYPE   PREEMPTIBLE  INTERNAL_IP  EXTERNAL_IP      STATUS
e2e-test-jsz-master             europe-west1-b  n1-standard-1               10.240.0.2   104.199.90.204   RUNNING
e2e-test-jsz-minion-group-09c3  europe-west1-b  n1-standard-2               10.240.0.4   146.148.115.195  RUNNING
e2e-test-jsz-minion-group-p8g9  europe-west1-b  n1-standard-2               10.240.0.3   104.199.22.157   RUNNING
e2e-test-jsz-minion-group-rqt7  europe-west1-b  n1-standard-2               10.240.0.5   104.199.31.245   RUNNING
```

```
$ ./cluster/kubectl.sh get nodes
NAME                             STATUS                     AGE       VERSION
e2e-test-jsz-master              Ready,SchedulingDisabled   18h       v1.6.0-alpha.0.2942+e827393fa2335c-dirty
e2e-test-jsz-master-54d          Ready,SchedulingDisabled   17h       v1.6.0-alpha.0.2942+e827393fa2335c-dirty
e2e-test-jsz-master-ab1          Ready,SchedulingDisabled   17h       v1.6.0-alpha.0.2942+e827393fa2335c-dirty
e2e-test-jsz-minion-group-09c3   Ready                      18h       v1.6.0-alpha.0.2942+e827393fa2335c-dirty
e2e-test-jsz-minion-group-3m6v   Ready                      17h       v1.6.0-alpha.0.2942+e827393fa2335c-dirty
e2e-test-jsz-minion-group-3vwc   Ready                      17h       v1.6.0-alpha.0.2942+e827393fa2335c-dirty
e2e-test-jsz-minion-group-8jz8   Ready                      17h       v1.6.0-alpha.0.2942+e827393fa2335c-dirty
e2e-test-jsz-minion-group-hk66   Ready                      17h       v1.6.0-alpha.0.2942+e827393fa2335c-dirty
e2e-test-jsz-minion-group-j1j5   Ready                      17h       v1.6.0-alpha.0.2942+e827393fa2335c-dirty
e2e-test-jsz-minion-group-p8g9   Ready                      18h       v1.6.0-alpha.0.2942+e827393fa2335c-dirty
e2e-test-jsz-minion-group-rqt7   Ready                      18h       v1.6.0-alpha.0.2942+e827393fa2335c-dirty
e2e-test-jsz-minion-group-s1m5   NotReady                   17h       v1.6.0-alpha.0.2942+e827393fa2335c-dirty
```

**What you expected to happen**:

Nodes are removed after their VM is removed.

**How to reproduce it** (as minimally and precisely as possible):

You can run any of HA master e2e tests: 
```
test/e2e/ha_master.go:
It("survive addition/removal replicas same zone [Serial][Disruptive]" ...
It("survive addition/removal replicas different zones [Serial][Disruptive]" ...
It("survive addition/removal replicas multizone workers [Serial][Disruptive]" ...
```

**Anything else do we need to know**:

Logs of node controller:
```
E0111 08:36:26.507186       5 nodecontroller.go:781] Error updating node e2e-test-jsz-minion-group-3m6v: User "system:serviceaccount:kube-system:node-controller" cannot update nodes at the cluster scope. (put nodes e2e-test-jsz-minion-group-3m6v)
I0111 08:36:26.578560       5 nodecontroller.go:742] node e2e-test-jsz-minion-group-3m6v hasn't been updated for 17h15m14.620822227s. Last ready condition is: {Type:Ready Status:True LastHeartbeatTime:2017-01-10 15:13:58 +0000 UTC LastTransitionTime:2017-01-10 14:28:49 +0000 UTC Reason:KubeletReady Message:kubelet is posting ready status. WARNING: CPU hardcapping unsupported}
I0111 08:36:26.578646       5 nodecontroller.go:769] node e2e-test-jsz-minion-group-3m6v hasn't been updated for 17h15m14.620915183s. Last out of disk condition is: &NodeCondition{Type:OutOfDisk,Status:False,LastHeartbeatTime:2017-01-10 15:13:58 +0000 UTC,LastTransitionTime:2017-01-10 14:28:11 +0000 UTC,Reason:KubeletHasSufficientDisk,Message:kubelet has sufficient disk space available,}
E0111 08:36:26.607364       5 nodecontroller.go:781] Error updating node e2e-test-jsz-minion-group-3m6v: User "system:serviceaccount:kube-system:node-controller" cannot update nodes at the cluster scope. (put nodes e2e-test-jsz-minion-group-3m6v)
E0111 08:36:26.707167       5 nodecontroller.go:500] Unable to mark all pods NotReady on node e2e-test-jsz-minion-group-3m6v: User "system:serviceaccount:kube-system:node-controller" cannot list pods at the cluster scope. (get pods)
I0111 08:36:26.883243       5 nodecontroller.go:513] Deleting node (no longer present in cloud provider): e2e-test-jsz-minion-group-3m6v
E0111 08:36:26.885863       5 nodecontroller.go:521] Unable to forcefully delete node "e2e-test-jsz-minion-group-3m6v": unable to delete node "e2e-test-jsz-minion-group-3m6v": User "system:serviceaccount:kube-system:node-controller" cannot delete nodes at the cluster scope. (delete nodes e2e-test-jsz-minion-group-3m6v)
```

**Comment by jszczepkowski** [comment]
CC @gmarek

**Comment by gmarek** [comment]
The culprit here is 
```
"system:serviceaccount:kube-system:node-controller" cannot delete nodes at the cluster scope. (delete nodes e2e-test-jsz-minion-group-3m6v)
```
which sounds like an RBAC problem.

**Comment by gmarek** [comment]
Also - is this problem present in 1.5 release?

**Comment by deads2k** [comment]
> Also - is this problem present in 1.5 release?

No, RBAC enforcment started in 1.6.  Looks like @liggitt fixed this up here: https://github.com/kubernetes/kubernetes/commit/bda95a59adad71146104a61290957449d4d2a73d#diff-a066255fca075e2bdcfe045e7ca352f7

**Comment by yujuhong** [comment]
> No, RBAC enforcment started in 1.6. Looks like @liggitt fixed this up here: bda95a5#diff-a066255fca075e2bdcfe045e7ca352f7

Yes, this is a dup of #39639, and the fix already went in.
